File: TAbaretata/CSharpPartTwo-2015/Multidimensional Arrays/04.BinarySearch/BinarySearch.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

//Write a program, that reads from the console an array of N integers and an integer K,
//sorts the array and using the method Array.BinSearch() finds the largest number in the array which is ≤ K. 

class BinarySearch
{
    static void Main()
    {
        Console.Write("Enter length of the array: ");
        int n = int.Parse(Console.ReadLine());
        Console.Write("Enter max value: ");
        int k = int.Parse(Console.ReadLine());
        int[] arr = new int[n];
        int maxValue = 0;
        for (int i = 0; i < n; i++)
        {
            Console.Write("arr[{0}]= ", i);
            arr[i] = int.Parse(Console.ReadLine());
        }
        Array.Sort(arr);
        for (int i = 0; i < n; i++)
        {

            if (arr[i] < k)
            {
                maxValue = arr[i];
            }
        }
        Array.BinarySearch(arr, maxValue);
        if (arr[0] > k)
        {
            Console.WriteLine("No number is lesser than the given max value.");
        }
        else
        {
            Console.WriteLine("Max number lesser than the given max value is: {0}", maxValue);
        }
    }
}

File: TAbaretata/JavaScript-Fundamentals-2015/05.Arrays/07.BinarySearch/BinarySearch.js

// Problem 7: Write a program that finds the index of given element in a sorted array of integers by using
// the binary search algorithm 

var arr = [3, 1, 5, 12, 4, 8, 2];
var key = 4,
    first = 0,
    last = arr.length - 1;

arr.sort(function (a, b) {
    return a - b;
});

while (last >= first) {
    var mid = parseInt((first + last) / 2);
    if (arr[mid] < key) {
        first = mid + 1;
    } else if (arr[mid] > key) {
        last = mid - 1;
    } else {
        console.log(mid);
        break;
    }
}